src/read.c

```diff
@@ -1771,25 +1799,29 @@ static avifBool avifParseAV1CodecConfigurationBox(const uint8_t * raw, size_t ra
 
     if (markerAndVersion != 0x81) {
         // Marker and version must both == 1
-        avifDiagnosticsPrintf(diag, "av1C contains illegal marker and version pair: [%u]", markerAndVersion);
+        avifDiagnosticsPrintf(diag, "%s contains illegal marker and version pair: [%u]", configPropName, markerAndVersion);
         return AVIF_FALSE;
     }
 
-    av1C->seqProfile = (seqProfileAndIndex >> 5) & 0x7;    // unsigned int (3) seq_profile;
-    av1C->seqLevelIdx0 = (seqProfileAndIndex >> 0) & 0x1f; // unsigned int (5) seq_level_idx_0;
-    av1C->seqTier0 = (rawFlags >> 7) & 0x1;                // unsigned int (1) seq_tier_0;
-    av1C->highBitdepth = (rawFlags >> 6) & 0x1;            // unsigned int (1) high_bitdepth;
-    av1C->twelveBit = (rawFlags >> 5) & 0x1;               // unsigned int (1) twelve_bit;
-    av1C->monochrome = (rawFlags >> 4) & 0x1;              // unsigned int (1) monochrome;
-    av1C->chromaSubsamplingX = (rawFlags >> 3) & 0x1;      // unsigned int (1) chroma_subsampling_x;
-    av1C->chromaSubsamplingY = (rawFlags >> 2) & 0x1;      // unsigned int (1) chroma_subsampling_y;
-    av1C->chromaSamplePosition = (rawFlags >> 0) & 0x3;    // unsigned int (2) chroma_sample_position;
+    config->seqProfile = (seqProfileAndIndex >> 5) & 0x7;    // unsigned int (3) seq_profile;
+    config->seqLevelIdx0 = (seqProfileAndIndex >> 0) & 0x1f; // unsigned int (5) seq_level_idx_0;
+    config->seqTier0 = (rawFlags >> 7) & 0x1;                // unsigned int (1) seq_tier_0;
+    config->highBitdepth = (rawFlags >> 6) & 0x1;            // unsigned int (1) high_bitdepth;
+    config->twelveBit = (rawFlags >> 5) & 0x1;               // unsigned int (1) twelve_bit;
+    config->monochrome = (rawFlags >> 4) & 0x1;              // unsigned int (1) monochrome;
+    config->chromaSubsamplingX = (rawFlags >> 3) & 0x1;      // unsigned int (1) chroma_subsampling_x;
+    config->chromaSubsamplingY = (rawFlags >> 2) & 0x1;      // unsigned int (1) chroma_subsampling_y;
+    config->chromaSamplePosition = (rawFlags >> 0) & 0x3;    // unsigned int (2) chroma_sample_position;
     return AVIF_TRUE;
 }
 
-static avifBool avifParseAV1CodecConfigurationBoxProperty(avifProperty * prop, const uint8_t * raw, size_t rawLen, avifDiagnostics * diag)
+static avifBool avifParseCodecConfigurationBoxProperty(avifProperty * prop,
+                                                       const uint8_t * raw,
+                                                       size_t rawLen,
+                                                       const char * configPropName,
+                                                       avifDiagnostics * diag)
 {
-    return avifParseAV1CodecConfigurationBox(raw, rawLen, &prop->u.av1C, diag);
+    return avifParseCodecConfigurationBox(raw, rawLen, &prop->u.av1C, configPropName, diag);
```

Comment:
nit (for a separate change): This function isn't doing anything. we can remove it and replace the callers directly.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1372